internal/util/filepath.go

```diff
@@ -0,0 +1,13 @@
+package util
+
+import "path/filepath"
+
+// ExtractDirPath removes the file part of a path if it exists.
+func ExtractDirPath(p string) string {
+	// If the base of the path has an extension, it's likely a file.
+	if filepath.Ext(filepath.Base(p)) != "" {
```

Comment:
I think it's likely that people will use a file without extension, e.g. `/etc/alloy_config` and this will not work for them.

I think instead we should `Stat` the path and check if it's directory.

Reply:
I did the change: https://github.com/grafana/alloy/pull/1726/commits/3f08a9ef7bc1b75745a8b4bd885a34f57d11f4a5. The error should not happen because if the path is wrong then it will fail earlier but we use this func everywhere for testing so I only logged it with warn